### DISCUSSION TITLE
Binary data ingestion

### DISCUSSION THREAD

**Discussion by vbhavsar** [opensearch-project/data-prepper#2807] [Ideas]
Currently, data-prepper is primarily focused on handling text-based formats. I would like to propose the expansion of data-prepper's capabilities to support binary data ingestion, specifically the inclusion of popular binary serialization formats such as Protocol Buffers (protobuf) and Apache Thrift.

This enhancement would open up new possibilities for building versatile data pipelines that can seamlessly integrate with diverse systems and applications.

**Comment by dlvenable** [comment]
@vbhavsar , Thank you for putting this proposal in. Do you have any specific use-cases in mind? Are you looking for a different set of processors as well? Would Data Prepper be receiving these over HTTP or are you also looking for TCP messaging?

**Reply by vbhavsar** [reply]
Hi @dlvenable. Thank you for responding. The use case I had in mind was for reading binary data from Kafka. It looks like there already is a [Kafka plugin](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/kafka-plugins).
